src/IpPort.cpp

```diff
@@ -8,28 +8,67 @@ std::ostream &operator<<(std::ostream &os, const IpPort &p)
     return os;
 }
 
+std::map<const uint16_t, PortData> IpPort::ports_tcp_list;
+std::map<const uint16_t, PortData> IpPort::ports_udp_list;
+
+void IpPort::set_csv_iana_ports(std::string path)
+{
+    io::CSVReader<3> in(path);
+    in.read_header(io::ignore_extra_column, "Service Name", "Port Number", "Transport Protocol");
+    std::string service, ports, protocol;
+    for (;;) {
+        try {
+            while (in.read_row(service, ports, protocol)) {
+                if (service.empty() || ports.empty() || !(protocol == "tcp" || protocol == "udp")) {
+                    continue;
+                }
+                auto delimiter = ports.find('-');
+                if (delimiter != ports.npos) {
+                    auto first_value = static_cast<uint16_t>(std::stoul(ports.substr(0, delimiter)));
+                    auto last_value = static_cast<uint16_t>(std::stoul(ports.substr(delimiter + 1)));
+                    if (protocol == "tcp") {
+                        ports_tcp_list[last_value] = {service, first_value};
+                    } else if (protocol == "udp") {
+                        ports_udp_list[last_value] = {service, first_value};
+                    }
+                } else {
+                    auto value = static_cast<uint16_t>(std::stoul(ports));
+                    if (protocol == "tcp") {
+                        ports_tcp_list[value] = {service, value};
+                    } else if (protocol == "udp") {
+                        ports_udp_list[value] = {service, value};
+                    }
+                }
+            }
+            break;
+        } catch ([[maybe_unused]] std::exception &e) {
+            in.next_line();
```

Comment:
should we warning here?

Reply:
I don't think so because it happens quite often on iana file due to multiline description for Ports like this:

```
virtualdj,,,VirtualDJ Remote Control protocol,[Atomix_Productions],[Stephane_Clavel],2011-10-12,,,,,Defined TXT keys: None
visel,,,visel Q-System services,[Lehpaner_Sasa],[Lehpaner_Sasa],,,,,,"Defined TXT keys:   Version=<service version>
  Type=<service type>
  Name=<service Name>
  Zone=<service zone>
  ID=<serviceID>
  SString=<serviceString>
  SString1=<serviceString>"
voalte2,,tcp,Server location via DNS-SD,[Voalte_Inc],[John_Simpson],2019-01-22,,,,,"Defined TXT keys: txtvers, name, (others not publicly documented)"
voalte3,,tcp,Server location via DNS-SD,[Voalte_Inc],[John_Simpson],2019-01-22,,,,,"Defined TXT keys: txtvers, name, (others not publicly documented)"
volterio,,tcp,Internal Communication Service,[Volterio],[Johannes_Ofner],2021-07-23,,,,,Defined TXT keys: None
vos,,,Virtual Object System (using VOP/TCP),[Reed_Hedges][Peter_Amstutz],[Reed_Hedges][Peter_Amstutz],,,,,,"Defined TXT keys:   url=<Object URL with full path>
  type=<Comma-seperated list of object types>
```

Reply:
So the log would be noisy